bigframes/session/clients.py

```diff
@@ -98,39 +99,65 @@ def __init__(
         self._location = location
         self._use_regional_endpoints = use_regional_endpoints
         self._credentials = credentials
+        self._bq_kms_key_name = bq_kms_key_name
 
         # cloud clients initialized for lazy load
         self._bqclient = None
+        self._bqmlclient = None
         self._bqconnectionclient = None
         self._bqstoragereadclient = None
         self._cloudfunctionsclient = None
         self._resourcemanagerclient = None
 
+    def _create_bigquery_client(self):
+        bq_options = None
+        if self._use_regional_endpoints:
+            bq_options = google.api_core.client_options.ClientOptions(
+                api_endpoint=(
+                    _BIGQUERY_REGIONAL_ENDPOINT
+                    if self._location.lower() in _REP_SUPPORTED_REGIONS
+                    else _BIGQUERY_LOCATIONAL_ENDPOINT
+                ).format(location=self._location),
+            )
+        bq_info = google.api_core.client_info.ClientInfo(
+            user_agent=self._application_name
+        )
+
+        bq_client = bigquery.Client(
+            client_info=bq_info,
+            client_options=bq_options,
+            credentials=self._credentials,
+            project=self._project,
+            location=self._location,
+        )
+
+        return bq_client
+
     @property
     def bqclient(self):
         if not self._bqclient:
-            bq_options = None
-            if self._use_regional_endpoints:
-                bq_options = google.api_core.client_options.ClientOptions(
-                    api_endpoint=(
-                        _BIGQUERY_REGIONAL_ENDPOINT
-                        if self._location.lower() in _REP_SUPPORTED_REGIONS
-                        else _BIGQUERY_LOCATIONAL_ENDPOINT
-                    ).format(location=self._location),
+            self._bqclient = self._create_bigquery_client()
+            if self._bq_kms_key_name:
+                self._bqclient.default_query_job_config = bigquery.QueryJobConfig(
+                    destination_encryption_configuration=bigquery.EncryptionConfiguration(
+                        kms_key_name=self._bq_kms_key_name
+                    )
+                )
+                self._bqclient.default_load_job_config = bigquery.LoadJobConfig(
+                    destination_encryption_configuration=bigquery.EncryptionConfiguration(
+                        kms_key_name=self._bq_kms_key_name
+                    )
                 )
-            bq_info = google.api_core.client_info.ClientInfo(
-                user_agent=self._application_name
-            )
-            self._bqclient = bigquery.Client(
-                client_info=bq_info,
-                client_options=bq_options,
-                credentials=self._credentials,
-                project=self._project,
-                location=self._location,
-            )
 
         return self._bqclient
 
+    @property
+    def bqmlclient(self):
```

Comment:
Why we need separate clients for BQ and BQML?

Reply:
Oh saw your comment

Reply:
Well still, it doesn't need two separate clients. Instead we can do sth different in BQML with the same bqclient. The simplest way, _start_query() and _start_query_bqml() doesn't need to target at 2 different clients. (Also IMO it is better to have only one method with parameters.) Further more, we may extract the needed bqclient methods in a BqClientManager, and for BQML, there can be a derived class from BqClientManager which overrides start_query method.

It will help us keep the clients in sync.

Reply:
As discussed with you offline, sent a patch to:
1. Reverted the bqmlclient separate from bqclient
2. Moved setting the encryption key from default job config to call time job config 

This comes up with the trade-off that now developer and reviewers need to make sure the code paths are setting the cmek (if any) correctly. In this change we are adding common methods to help with that:
1. _prepare_query_job_config
2. _prepare_load_job_config
3. _prepare_job_job_config